lib/logger.rb

```diff
@@ -380,7 +380,9 @@ class Logger
   include Severity
 
   # Logging severity threshold (e.g. <tt>Logger::INFO</tt>).
-  attr_reader :level
+  def level
+    @tll[Thread.current] || @level
```

Comment:
Using `Thread.current` as a key is different from using `Thread.current[key]` as the former is per-thread and the latter is per-fiber.

```suggestion
    @level_override[Fiber.current] || @level
```

I also don't like `@tll` as it's not clear what that means. I'd call it what it is, "level_override" or something like that.

Reply:
thread local level was my guess, but I agree a more clear name would be optimal.  However, that change could be made later.

Ideally there would be a method returning the key to use, so you could easily subclass and override to get Fiber-local, Thread-local, or global behavior.  I don't have a problem defaulting to Fiber-local.

Reply:
I'm personally okay with not having a public interface for this yet, as I want to propose some further changes as you suggested. I think if we can start with fiber local, it's good, and we can later expand that. It should be good enough for @mperham use case (equivalent to what's currently in Sidekiq).